tests/suites/test_suite_psa_crypto.data

```diff
@@ -5424,10 +5424,26 @@ PSA key derivation: TLS 1.2 Mix-PSK-to-MS, other key is raw data
 depends_on:PSA_WANT_ALG_SHA_256:PSA_WANT_ALG_TLS12_PSK_TO_MS
 derive_output:PSA_ALG_TLS12_PSK_TO_MS(PSA_ALG_SHA_256):PSA_KEY_DERIVATION_INPUT_SEED:"5bc0b19b4a8b24b07afe7ec65c471e94a7d518fcef06c3574315255c52afe21b5bc0b19b872b9b26508458f03603744d575f463a11ae7f1b090c012606fd3e9f":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_OTHER_SECRET:"c4eb02cb10175ab8a33aeeb068ba23df08206b0e":PSA_ERROR_INVALID_ARGUMENT:PSA_KEY_DERIVATION_INPUT_SECRET:"01020304":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_LABEL:"6d617374657220736563726574":PSA_SUCCESS:"":48:"2d6cba7ad474ac3dfcc01516ed7161a9b00c2e8a35b3e921b9eb09b546a54c98491b21d1baafb659c9094b760144ea1f":"":11:1:0
 
-PSA key derivation: TLS 1.2 Mix-PSK-to-MS, output key not permitted
+# output key test: secret passed as key, other secret passed as key
+PSA key derivation: TLS 1.2 Mix-PSK-to-MS, output key ok #1
+depends_on:PSA_WANT_ALG_SHA_256:PSA_WANT_ALG_TLS12_PSK_TO_MS
+derive_output:PSA_ALG_TLS12_PSK_TO_MS(PSA_ALG_SHA_256):PSA_KEY_DERIVATION_INPUT_SEED:"5bc0b19b4a8b24b07afe7ec65c471e94a7d518fcef06c3574315255c52afe21b5bc0b19b872b9b26508458f03603744d575f463a11ae7f1b090c012606fd3e9f":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_OTHER_SECRET:"c4eb02cb10175ab8a33aeeb068ba23df08206b0e":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_SECRET:"01020304":PSA_SUCCESS:PSA_KEY_DERIVATION_INPUT_LABEL:"6d617374657220736563726574":PSA_SUCCESS:"":48:"2d6cba7ad474ac3dfcc01516ed7161a9b00c2e8a35b3e921b9eb09b546a54c98491b21d1baafb659c9094b760144ea1f":"":1:1:1
+
+# output key test: secret passed as key, other secret passed as bytes
```

Comment:
Hmm, I'm not sure that's the behaviour we want. IIUC, `output_key()` should only work when _all_ secret inputs (in that case, but "secret" and "other secret") have been passed securely, where "securely" means either `input_key()` or `key_derivation_key_agreement()`, but not `input_bytes()`.

So, I think this should be a "not permitted" case (so, the code would need to be changed).

Also, we probably want to have positive test with `key_derivation_key_agreement()`.

@gilles-peskine-arm can you weigh in on this?

Reply:
The way it's currently specified, only the `INPUT_SECRET` matters for allowing `output_key`. The `INPUT_OTHER_SECRET` doesn't matter.

The rationale for restricting `output_key` is hygiene. It's to reduce the risk of creating a non-exportable key object and believing it to be impossible to reconstruct outside the PSA boundary, in scenarios where it can be reconstructed outside the PSA boundary because the KDF inputs are available outside. For this purpose, it is sufficient that one of the inputs is a PSA key (which has to be non-exportable for the reasoning to apply — a limitation of this hygiene restriction). In practice, this input is likely the PSK (long-term secret) and not the typically-public inputs or the single-use “other secret”.

Reply:
Oh, right, thanks for the explanation! So, I was wrong and the code is right - I'll update my review then.